Package.swift

```diff
@@ -1,34 +1,60 @@
-// swift-tools-version:5.6
+// swift-tools-version:5.7
 
 import PackageDescription
 
 let package = Package(
     name: "swift-service-lifecycle",
+    platforms: [
+        .macOS(.v10_15),
+        .iOS(.v13),
+        .watchOS(.v6),
+        .tvOS(.v13),
+    ],
     products: [
-        .library(name: "Lifecycle", targets: ["Lifecycle"]),
-        .library(name: "LifecycleNIOCompat", targets: ["LifecycleNIOCompat"]),
+        .library(
+            name: "ServiceLifecycle",
+            targets: ["ServiceLifecycle"]
+        ),
+        .library(
+            name: "UnixSignals",
+            targets: ["UnixSignals"]
+        ),
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-atomics.git", from: "1.0.0"),
-        .package(url: "https://github.com/apple/swift-log.git", from: "1.0.0"),
-        .package(url: "https://github.com/apple/swift-metrics.git", "1.0.0" ..< "3.0.0"),
-        .package(url: "https://github.com/swift-server/swift-backtrace.git", from: "1.1.1"),
-        .package(url: "https://github.com/apple/swift-nio.git", from: "2.0.0"), // used in tests
-        .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
```

Comment:
emitting metric on application start and shutdown has been _very_ valuable in real world production services, the canonical example is a monitoring this to detect an a spike in application restart because of crashes

Reply:
Generally I agree. But ServiceLifecycle created the weird situation in the past where ServiceLifecycle depended on a MetricSystem that adopters wanted to setup with ServiceLifecycle. Basically the first use occurred even before the system was actually ready to create Counters. A good example for this are MetricSystems that are sending their data to a server themselves.

Reply:
I can see your point. Maybe the solution is for this library to take an abstract handler that can emit the specific metrics, or even something like a delegate so that the application / service can do the bridging

Reply:
Following up on this, I would like to get the PR merged without metrics and then add it on top since the PR is getting quite big.